tests/Doctrine/Tests/ORM/Functional/Ticket/GH7805Test.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\ORM\Tools\Pagination\Paginator;
+use Doctrine\Tests\Models\CMS\CmsArticle;
+use Doctrine\Tests\OrmFunctionalTestCase;
+use function count;
+use function iterator_to_array;
+
+class GH7805Test extends OrmFunctionalTestCase
+{
+    protected function setUp()
+    {
+        $this->useModelSet('cms');
+        parent::setUp();
+    }
+
+    public function testPaginationWithSimpleArithmetic()
+    {
+        $article = new CmsArticle();
+
+        $article->topic   = 'Test SimpleArithmetic ORDER BY';
+        $article->text    = 'This test fails on MySQL.';
+        $article->version = 1;
+
+        $this->_em->persist($article);
+        $this->_em->flush();
+
+        $query = $this->_em->createQuery('SELECT a FROM Doctrine\Tests\Models\CMS\CmsArticle a ORDER BY a.version + 0 ASC');
```

Comment:
Does this now also work with custom DQL functions?

Reply:
Originally, only `PathExpression` found in the `ORDER BY` part was added to the `SELECT`, everything else was skipped.

I just added another condition to also accept `SimpleArithmetic` (see line 102 upper in this file) so i don't think custom DQL functions will work unless they are considered as `PathExpression` or `SimpleArithmetic`.

Maybe we should consider that every `Node` found in the `ORDER BY` part could be added in the `SELECT` part and remove the condition on line 102?

Reply:
I'm not sure why the `Identity` function wrapper was added in first place, but if you could add an SQL generation code using something like `SQRT()`, that would be sufficient to check how we need to modify the code.

Reply:
Done